files/image_config/logrotate/logrotate-config.sh

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+
+# Reserve space for btmp and wtmp, auth.log and cron.log, as well as logs that
+# should be disabled, just in case they get created and rotated
+RESERVED_SPACE_KB=1024
+
+# Number of rotated archive files to keep per log file
+NUM_ARCHIVES_PER_LOG_FILE=7
+
+# Number of files to rotate in each block within rsyslog.j2 template
+NUM_LOG_FILES_SYSLOG=1
+NUM_LOG_FILES_QUAGGA_TEAMD=3
+NUM_LOG_FILES_SWSS=2
+
+# Percentage of usable /var/log/ space to allocate to each block in template
+# NOTE: These must sum to 100!
+PERCENT_ALLOCATED_SYSLOG=75
+PERCENT_ALLOCATED_QUAGGA_TEAMD=15
+PERCENT_ALLOCATED_SWSS=10
+
+VAR_LOG_SIZE_KB=$(df /var/log | sed -n 2p | awk '{print $2}')
+USABLE_SPACE_KB=$((VAR_LOG_SIZE_KB - RESERVED_SPACE_KB))
+
+SIZE_LIMIT_SYSLOG_FILES_KB=$(((USABLE_SPACE_KB * PERCENT_ALLOCATED_SYSLOG / 100) / (NUM_LOG_FILES_SYSLOG * (NUM_ARCHIVES_PER_LOG_FILE + 1))))
+SIZE_LIMIT_QUAGGA_TEAMD_FILES_KB=$(((USABLE_SPACE_KB * PERCENT_ALLOCATED_QUAGGA_TEAMD / 100) / (NUM_LOG_FILES_QUAGGA_TEAMD * (NUM_ARCHIVES_PER_LOG_FILE + 1))))
+SIZE_LIMIT_SWSS_FILES_KB=$(((USABLE_SPACE_KB * PERCENT_ALLOCATED_SWSS / 100) / (NUM_LOG_FILES_SWSS * (NUM_ARCHIVES_PER_LOG_FILE + 1))))
```

Comment:
what are the actual sizes when the log partition size is 100M?

for example syslog log size limit is 102400 * 75 / 100 / (1 * ( 7 + 1) = 9600KB.
swss size limit is 102400*10/100/(2*(7+1)) = 640KB

since, the gzipped archive is much smaller than the actual log size. we might end up with not fully utilizing the /var/log partition.

Ideally, if we can rotate based on the free disk size, we can utilize the /var/log partition. however, we need to modify logrotate for that.

there are some discussion in here. https://serverfault.com/questions/372809/free-space-driven-log-rotation-on-linux